spec/brainstem/concerns/controller_dsl_spec.rb

```diff
@@ -489,6 +489,137 @@ module Concerns
+        
+        it "sets the correct configuration" do
+          subject.brainstem_params do
+            valid_dynamic_param :integer,
+              info: "User ID is required",
+              required: true
+          end
+
+          valid_params = subject.configuration[:_default][:valid_params]
+          expect(valid_params.keys.length).to eq(1)
+          expect(valid_params.keys[0]).to be_a(Proc)
+          expect(valid_params.keys[0].call).to eq(dynamic_keyword)
+
+          sprocket_ids_config = valid_params[valid_params.keys[0]]
+          expect(sprocket_ids_config[:info]).to eq "User ID is required"
+          expect(sprocket_ids_config[:required]).to be_truthy
+          expect(sprocket_ids_config[:type]).to eq("integer")
+          expect(sprocket_ids_config[:dynamic_key]).to be_truthy
+        end
+
+        context "when type is hash" do
+          it "adds the nested fields to valid params" do
+            subject.brainstem_params do
+              valid :id, :integer
+
+              valid :info, :hash, required: true do |param|
+                param.valid_dynamic_param :string, required: true
+
+                param.valid :data, :hash do |double_nested_param|
+                  double_nested_param.valid_dynamic_param :string
+                end
+              end
+            end
+
+            valid_params = subject.configuration[:_default][:valid_params]
+            param_keys = valid_params.keys
+            expect(param_keys.length).to eq(5)
+
+            expect(param_keys[0].call).to eq('id')
+            id_config = valid_params[param_keys[0]]
+            expect(id_config[:root]).to be_nil
+            expect(id_config[:ancestors]).to be_nil
+
+            expect(param_keys[1].call).to eq('info')
+            info_key = param_keys[1]
+            info_config = valid_params[info_key]
+            expect(info_config[:root]).to be_nil
+            expect(info_config[:ancestors]).to be_nil
+
+            expect(param_keys[2].call).to eq(dynamic_keyword)
+            dynamic_info_nested_config = valid_params[param_keys[2]]
+            expect(dynamic_info_nested_config[:root]).to be_nil
+            expect(dynamic_info_nested_config[:ancestors]).to eq([info_key])
+            expect(dynamic_info_nested_config[:dynamic_key]).to be_truthy
+            expect(dynamic_info_nested_config[:required]).to be_truthy
+
+            expect(param_keys[3].call).to eq('data')
+            details_data_key = param_keys[3]
+            details_data_config = valid_params[details_data_key]
+            expect(details_data_config[:root]).to be_nil
+            expect(details_data_config[:ancestors]).to eq([info_key])
+
+            expect(param_keys[4].call).to eq(dynamic_keyword)
+            dynamic_data_nested_config = valid_params[param_keys[4]]
+            expect(dynamic_data_nested_config[:root]).to be_nil
+            expect(dynamic_data_nested_config[:ancestors]).to eq([info_key, details_data_key])
```

Comment:
Any reason we don't check the :dynamic_key on the config for the dynamic_data_nested_config?